reVX/least_cost_xmission/trans_cap_costs.py

```diff
@@ -260,42 +258,74 @@ def _parse_config(xmission_config=None):
 
         return xmission_config
 
-    @staticmethod
-    def _clip_costs(cost_fpath, cost_layer, row_slice, col_slice,
+    def _clip_costs(self, cost_layers, length_invariant_cost_layers=None,
                     barrier_mult=100):
-        """
-        Extract clipped cost arrays from exclusion .h5 files
+        """Extract clipped cost arrays from exclusion .h5 files
 
         Parameters
         ----------
-        cost_fpath : str
-            Full path of .h5 file with cost arrays
-        cost_layer : str
-            Name of cost layer to extract
-        row_slice : slice
-            slice along axis 0 (rows) to clip costs too
-        col_slice : slice
-            slice along axis 1 (columns) to clip costs too
+        cost_layers : List[str]
+            List of layers in H5 that are summed to determine total
+            costs raster used for routing. Costs and distances for each
+            individual layer are also reported (e.g. wet and dry costs).
+            deteremining path using main cost layer.
+        length_invariant_cost_layers : List[str] | None, optional
+            List of layers in H5 to be added to the cost raster. The
+            costs specified by these layers are not scaled with distance
+            traversed across the cell (i.e. fixed one-time costs for
+            crossing these cells).
         barrier_mult : int, optional
-            Multiplier on transmission barrier costs, by default 100
-
-        Returns
-        -------
-        cost : ndarray
-            2d clipped array of raw tie-line costs
-        mcp_cost : ndarray
-            2d clipped array of mcp cost = cost * barrier * barrier_mult
+            Multiplier on transmission barrier costs.
+            By default, ``100``.
         """
-        with ExclusionLayers(cost_fpath) as f:
-            cost = f[cost_layer, row_slice, col_slice]
-            barrier = f['transmission_barrier', row_slice, col_slice]
-        mcp_cost = cost + cost * barrier * barrier_mult
-        mcp_cost = np.where(mcp_cost < 0, -1, mcp_cost)
+        li_cost_layers = length_invariant_cost_layers or []
+        logger.debug("Building cost layer with the folowing inputs:"
+                     f"\n\t- cost_layers: {cost_layers}"
+                     f"\n\t- length_invariant_cost_layers: {li_cost_layers}"
+                     f"\n\t- barrier_mult: {barrier_mult}")
+        self._cost = np.zeros(self.clip_shape, dtype=np.float32)
+        overlap = np.zeros(self.clip_shape, dtype=np.uint8)
+        with ExclusionLayers(self._cost_fpath) as f:
+            for cost_layer in cost_layers:
+                cost = f[cost_layer, self._row_slice, self._col_slice]
+                self._cost += cost
```

Comment:
@ppinchuk , do you think there is a significant performance hit for doing this every time?

Reply:
It should not be significant, since we are only reading in a small portion of the array.

But even if it was, what do you think would be a good alternative here? If we read in the whole array, we may end up dealing with even more memory issues than we are already. As such, I am happy to trade a little bit of CPU time to save memory. Thoughts on that?

Reply:
Sounds good to me, I briefly forgot about the MCP memory bottleneck